src/runtime/type-variable-info.ts

```diff
@@ -21,9 +21,9 @@ interface TypeVariableInfoLiteral {
 
 export class TypeVariableInfo {
   name: string;
-  _canWriteSuperset?: Type;
-  _canReadSubset?: Type;
-  _resolution?: Type;
+  _canWriteSuperset?: Type|null;
+  _canReadSubset?: Type|null;
```

Comment:
Is it necessary to have both `?` and `|null` ?

Reply:
Yes, they are distinct. 

Lots of discussion in https://github.com/microsoft/TypeScript/issues/7426 

It might be worth it to move to Maybe<T> or Optional<T> for explicit empty values.  WDYT @alxrsngrtn  ?